src/app/FunZone/simon/simon.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, OnInit, ViewChild } from "@angular/core"
+
+@Component({
+  selector: "app-simon",
+  templateUrl: "./simon.component.html",
+  styleUrls: ["./simon.component.scss"],
+})
+export class SimonComponent {
+  @ViewChild("topLeft") topLeft!: HTMLObjectElement
+  @ViewChild("bottomLeft") bottomLeft!: HTMLObjectElement
+  @ViewChild("bottomRight") bottomRight!: HTMLObjectElement
+  @ViewChild("topRight") topRight!: HTMLObjectElement
+
+  constructor() {
+    this.canClick = false
+    this.sequence = []
+    this.sequenceToGuess = []
+    this.score = 0
+  }
+  canClick: boolean
+  sequence: any[]
+  sequenceToGuess: any[]
+  score: number
+
+  async start() {
+    this.startPanels()
+    this.score = 0
+    this.sequence = [this.getRandomPanel()]
+    this.sequenceToGuess = [...this.sequence]
+
+    await this.runSequens()
+  }
+
+  startPanels() {}
+
+  async runSequens() {
+    this.canClick = false
+    for (const panel of this.sequence) {
+      await this.flash(panel)
+    }
+    this.canClick = true
+  }
+
+  panelClicked = async (panel: string) => {
+    if (!this.canClick) return
+    const exeptedPanel = this.sequenceToGuess.shift()
+    console.log(exeptedPanel)
+    if (
+      String(exeptedPanel?.nativeElement.className).includes(
```

Comment:
After you type `sequenceToGuess` I'm guessing you won't need the String constructor here